### ISSUE TITLE
[illumos] net/sslh missing from 2017Q2

### ISSUE THREAD

**Issue by sjorge** [TritonDataCenter/pkgsrc#12]

**Comment by sjorge** [comment]
Looks like it fails to build due to some errors!

These to patches seems to make it compile!

```patch
$NetBSD$

--- common.h.orig	2016-03-29 19:19:05.000000000 +0000
+++ common.h
@@ -69,7 +69,7 @@ enum connection_state {
 
 /* A 'queue' is composed of a file descriptor (which can be read from or
  * written to), and a queue for deferred write data */
-struct queue {
+struct _queue {
     int fd;
     void *begin_deferred_data;
     void *deferred_data;
@@ -84,7 +84,7 @@ struct connection {
     /* q[0]: queue for external connection (client);
      * q[1]: queue for internal connection (httpd or sshd);
      * */
-    struct queue q[2];
+    struct _queue q[2];
 };
 
 #define FD_CNXCLOSED    0
@@ -95,7 +95,7 @@ struct connection {
 /* common.c */
 void init_cnx(struct connection *cnx);
 int connect_addr(struct connection *cnx, int fd_from);
-int fd2fd(struct queue *target, struct queue *from);
+int fd2fd(struct _queue *target, struct _queue *from);
 char* sprintaddr(char* buf, size_t size, struct addrinfo *a);
 void resolve_name(struct addrinfo **out, char* fullname);
 void log_connection(struct connection *cnx);
@@ -110,8 +110,8 @@ int resolve_split_name(struct addrinfo *
 
 int start_listen_sockets(int *sockfd[], struct addrinfo *addr_list);
 
-int defer_write(struct queue *q, void* data, int data_size);
-int flush_deferred(struct queue *q);
+int defer_write(struct _queue *q, void* data, int data_size);
+int flush_deferred(struct _queue *q);
 
 extern int probing_timeout, verbose, inetd, foreground, 
        background, transparent, numeric;
```

```patch
$NetBSD$

--- common.c.orig	2016-03-29 19:19:05.000000000 +0000
+++ common.c
@@ -237,7 +237,7 @@ int connect_addr(struct connection *cnx,
 }
 
 /* Store some data to write to the queue later */
-int defer_write(struct queue *q, void* data, int data_size)
+int defer_write(struct _queue *q, void* data, int data_size)
 {
     char *p;
     if (verbose)
@@ -261,7 +261,7 @@ int defer_write(struct queue *q, void* d
  * Upon success, the number of bytes written is returned.
  * Upon failure, -1 returned (e.g. connexion closed)
  * */
-int flush_deferred(struct queue *q)
+int flush_deferred(struct _queue *q)
 {
     int n;
 
@@ -313,7 +313,7 @@ void dump_connection(struct connection *
  * returns FD_STALLED if data was read, could not be written, and has been
  * stored in temporary buffer.
  */
-int fd2fd(struct queue *target_q, struct queue *from_q)
+int fd2fd(struct _queue *target_q, struct _queue *from_q)
 {
    char buffer[BUFSIZ];
    int target, from, size_r, size_w;
@@ -594,7 +594,7 @@ void setup_syslog(const char* bin_name)
     int res;
 
     name1 = strdup(bin_name);
-    res = asprintf(&name2, "%s[%d]", basename(name1), getpid());
+    res = asprintf(&name2, "%s[%ld]", basename(name1), getpid());
     CHECK_RES_DIE(res, "asprintf");
     openlog(name2, LOG_CONS, LOG_AUTH);
     free(name1);
@@ -694,7 +694,7 @@ void write_pid_file(const char* pidfile)
         exit(3);
     }
 
-    fprintf(f, "%d\n", getpid());
+    fprintf(f, "%ld\n", getpid());
     fclose(f);
 }
 
```

They may need some minor cleanup, not sure if these break other platforms but it makes it word on illumos!

**Comment by sjorge** [comment]
@jperkin this may need a #ifdef, don't have a netbsd host to see if this breaks non illumos.

**Comment by jperkin** [comment]
Resolved as part of #14, will backport this to 2017Q2 when it shows up in git.
